src/core/cryptor.rs

```diff
@@ -0,0 +1,57 @@
+//! Cryptor module
+//!
+//! This module contains the [`Cryptor`] trait which is used to implement
+//! encryption and decryption of published data.
+
+use crate::core::error::PubNubError;
+
+/// This trait is used to encrypt and decrypt messages sent to the
+/// [`PubNub API`].
+///
+/// It is used by the [`dx`] modules to encrypt messages sent to PubNub and
+/// returned by the [`PubNub API`].
+///
+/// To implement this trait, you must provide `encrypt` and `decrypt` methods
+/// that takes a `&[u8]` and returns a `Result<Vec<u8>, PubNubError>`.
+///
+/// You can implement this trait for your own types, or use one of the provided
+/// features to use a crypto library.
+///
+/// # Examples
+/// ```
+/// use pubnub::core::{Cryptor, error::PubNubError};
+///
+/// struct MyCryptor;
+///
+/// impl Cryptor for MyCryptor {
+///     fn encrypt<'en, T: Into<&'en [u8]>>(&self, source: T) -> Result<Vec<u8>, PubNubError> {
```

Comment:
We use `where` syntax in all of the files.
```suggestion
///     fn encrypt<'en, T>(&self, source: T) -> Result<Vec<u8>, PubNubError> where T: Into<&'en [u8]> {
```

Reply:
@Xavrax I was struggling with whether I needed to do it or not. Because at least in docs they mentioned, that single bound is fine inline if it shorts, if more bound traits appear, they advised to use `where`. I will switch then.

Reply:
@Xavrax changed in the last commit.